### ISSUE TITLE
Changes to selected algorithms don't always save

### ISSUE THREAD

**Issue by vertigo220** [namazso/OpenHashTab#138]
I was trying to test the speed when calculating several vs a few hashes for a large file, but even after deselecting some, it took the same exact length of time. I then found that when I unchecked them and exited out then went back into options, they were rechecked. It kept doing it every time for a while, then when trying to replicate it a little later I couldn't for a while as they were saving properly, but then I got it to happen again. I still haven't been able to figure out how to consistently replicate it, and at first I was questioning whether to report, but since I've at least been able to get it to occur twice I figured I should.

**Comment by namazso** [comment]
some settings like "If clipboard contains a hash, automatically enable algorithms of that length for the run" modify the current run's configuration which will appear as such in the settings, rather than the actual setting value. I assume this is the problem you hit.

**Comment by vertigo220** [comment]
Yes, I'm pretty sure that was it. Not sure how best to handle this, maybe a prominent (red/bold text) warning, but there should really be something to make the user aware that this is a possibility and, if possible, _when_ it is happening. Maybe any algorithms that were automatically enabled due to this setting could be marked somehow with something indicating that they have been automatically enabled and why. As it was, I wasted several minutes attempting to run the initial speed test and another 20+ minutes attempting to troubleshoot what I thought was a very specific bug.
